docker/install/ubuntu_install_riscv_gcc_toolchain.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -e
+set -u
+set -o pipefail
+
+function show_usage() {
+    cat <<EOF
+Usage: docker/install/ubuntu_install_riscv_gcc_toolchain.sh <INSTALLATION_PATH>
+INSTALLATION_PATH is the installation path for the toolchain.
+EOF
+}
+
+if [ "$#" -lt 1 -o "$1" == "--help" -o "$1" == "-h" ]; then
+    show_usage
+    exit 1
+fi
+
+INSTALLATION_PATH=$1
+shift
+
+# Create installation path directory
+mkdir -p "${INSTALLATION_PATH}"
+
+# Download and extract RISC-V gcc
+TMPDIR=$(mktemp -d)
+RISCV_GCC_EXT="tar.gz"
+RLS_TAG="2023.07.07"
+RISCV_GCC_URL="https://github.com/riscv-collab/riscv-gnu-toolchain/releases/download/${RLS_TAG}/riscv64-glibc-ubuntu-22.04-gcc-nightly-${RLS_TAG}-nightly.${RISCV_GCC_EXT}"
```

Comment:
In contrast to the Xuantie GCC, this is a non-multilib build of the RISC-V GNU Toolchain. Support is probably limited to `rv64gc/lp64d`.

In the long term, we should find a way to support `riscv32` and `riscv64` (eventually, with different levels of floating-point support).

Reply:
Thanks @PhilippvK! I'll replace it with a multilib build.

Reply:
Do you have an guess on how long the from source build will take? I am not sure If this would be acceptable… Maybe there is a multilib build available to use here: https://github.com/sifive/freedom-tools/releases

Reply:
I guess it will take a few hours, by the link you specified there are builds from two years ago. As a solution, I can publish a release in forked riscv-gnu-toolchain repository.